mlflow/llama_index/__init__.py

```diff
@@ -140,11 +147,32 @@ def save_model(
     from mlflow.llama_index.serialize_objects import serialize_settings
 
     _validate_engine_type(engine_type)
+
+    # TODO: make this logic cleaner and maybe a util
+    with tempfile.TemporaryDirectory() as temp_dir:
+        index_or_code_path = (
+            _validate_and_get_model_code_path(index, temp_dir) if isinstance(index, str) else index
+        )
+
+        _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+        path = os.path.abspath(path)
+        _validate_and_prepare_target_save_path(path)
+
+        if isinstance(model_config, str):
+            model_config = _validate_and_get_model_config_from_file(model_config)
+
+        model_code_path = None
+        if isinstance(index_or_code_path, str):
```

Comment:
1. We need to add a condition check at L202 like [LangChain implementation](https://github.com/mlflow/mlflow/blob/master/mlflow/langchain/__init__.py#L342-L347), so we don't save the index object itself when the code path is given. Currently, it saves the serialized index object so does not provide the benefit of model-from-code.
2. Also we need to pass the code path to `pyfunc.add_to_model` at L204 so it can be loaded as pyfunc.